src/lib/course.ts

```diff
@@ -0,0 +1,87 @@
+import bent from 'bent'
+import config from './config'
+
+const getJson = bent('json')
+
+export type Assignment = {
+  id: string
+  name: string
+}
+
+export type Course = {
+  id: string
+  name: string
+  assignments: Assignment[] 
+}
+
+export type StudentProgress = {
+  student_id: string
+  student_email: string
+  seconds_spent: number
+  grade: number
+  status: string
+  completion_date: {
+    seconds: number
+  }
+  date: Date
+}
+
+export async function info(courseId: string): Promise<Course> {
+  if (!config) {
+    throw new Error('No Config')
+  }
+  try {
+    const token = config.getToken()
+    const domain = config.getDomain()
+    const authHeaders = {
+      'Authorization': `Bearer ${token}`
+    }
+
+    return getJson(`https://octopus.${domain}/api/v1/courses/${courseId}`, undefined, authHeaders)
+  } catch (error) {
+    if (error.json) {
+      const message = JSON.stringify(await error.json())
+      throw new Error(message)
+    }
+    throw error
+  }
+}
+
+function fixDate(progress: StudentProgress) {
+  if (progress.completion_date) {
+    const t = new Date(1970, 0, 1); // Epoch
+    t.setUTCSeconds(progress.completion_date.seconds);
```

Comment:
Maybe it's better to extract this to a helper function.

Reply:
fixed